### DISCUSSION TITLE
Modals Text Input

### DISCUSSION THREAD

**Discussion by malikmajai** [mccoderpy/discord.py-message-components#19] [Q&A] [answered]
The lib is fantastic, everything is there but why text input modals are not included yet? 
<img width="224" alt="Capture" src="https://user-images.githubusercontent.com/92770716/165101010-51e57b13-85b5-421a-9a0f-276d3f28b47c.PNG">

**Comment by mccoderpy** [comment, marked as answer]
> The lib is fantastic, everything is there but why text input modals are not included yet? 
<img alt="Capture" width="224" src="https://user-images.githubusercontent.com/92770716/165101010-51e57b13-85b5-421a-9a0f-276d3f28b47c.PNG">

Thank you 😄, I do what I can. 
Modals are implemented. But they are only in thee developer branch at the moment.(Like most of the other things too)

**Reply by malikmajai** [reply]
:) Oh that's great,

I really liked the library, When we can use Modals? Also including slash commands can fulfil all needs. 

And I really liked your hard work, everything is there in your lib. And the docs are also really nice.

**Reply by mccoderpy** [reply]
> :) Oh that's great,
> 
> I really liked the library, When we can use Modals? Also including slash commands can fulfil all needs. 
> 
> And I really liked your hard work, everything is there in your lib. And the docs are also really nice. 

If you install the developer-branch you already can use Modals, slash-commands & co.. The only problem is, that there are no external docs for them yet. But there are some examples in the examples folder
